packages/discovery-react-components/src/components/SearchResults/components/Result/__tests__/Result.test.tsx

```diff
@@ -3,743 +3,547 @@ import React from 'react';
-              extracted_metadata: {
-                filename: 'my file name'
-              },
-              myTitle: 'my title'
-            }
-          ];
-          const api = {
-            setSelectedResult: jest.fn()
-          };
-          const { getByText } = render(
-            wrapWithContext(<SearchResults resultTitleField="doesNotExist" />, api, context)
-          );
-          expect(getByText('my file name')).toBeInTheDocument();
+    describe('when there is a value for resultTitleField', () => {
+      beforeEach(() => {
+        componentProps.resultTitleField = 'myTitle';
+      });
+
+      describe('and there is a corresponding value on the QueryResult object', () => {
+        beforeEach(() => {
+          queryResults![0].myTitle = 'my title';
+        });
+
+        test('we display the value at that property', () => {
+          ({ searchResults } = setup(componentProps, { queryResults }));
+          expect(searchResults.getByText('my title')).toBeInTheDocument();
         });
       });
 
-      describe('and we have a value only for document_id', () => {
-        test('we display document_id', () => {
-          (context.searchResponseStore!.data as DiscoveryV2.QueryResponse).results = [
-            {
-              document_id: 'some document_id',
-              result_metadata: {
-                collection_id: '1'
-              },
-              myTitle: 'my title'
-            }
-          ];
-          const api = {
-            setSelectedResult: jest.fn()
-          };
-          const { getByText } = render(
-            wrapWithContext(<SearchResults resultTitleField="doesNotExist" />, api, context)
-          );
-          expect(getByText('some document_id')).toBeInTheDocument();
+      describe('but there is no corresponding value on the QueryResult object', () => {
+        beforeEach(() => {
+          queryResults![0].myTitle = undefined;
+        });
+
+        describe('but there is a value at extracted_metadata.title', () => {
+          beforeEach(() => {
+            queryResults![0].extracted_metadata = { title: 'some title' };
+          });
+
+          test('we display the value at extracted_metadata.title', () => {
+            ({ searchResults } = setup(componentProps, { queryResults }));
+            expect(searchResults.getByText('some title')).toBeInTheDocument();
+          });
+        });
+
+        describe('but there is a value for extracted_metadata.filename and document_id', () => {
+          beforeEach(() => {
+            queryResults![0].extracted_metadata = { filename: 'my file name' };
+            queryResults![0].document_id = 'some document_id';
+          });
+
+          test('displays the filename', () => {
+            ({ searchResults } = setup(componentProps, { queryResults }));
+            expect(searchResults.getByText('my file name')).toBeInTheDocument();
+          });
+        });
+
+        test('displays the value of document_id by default', () => {
```

Comment:
Should this get moved up to go around line 426 after the initial `beforeEach` in this `describe` block?